qiskit_metal/toolbox_metal/layer_stack_handler.py

```diff
@@ -0,0 +1,278 @@
+            return None
+
+        props = Dict()
+        thickness = 0.0
+        z_coord = 0.0
+        material = None
+        fill_value = None
+        chip_name = None
+
+        if self.ls_df is None:
+            abs_path = os.path.abspath(self.filename_csv_df)
+            self.logger.error(
+                f'Not able to read file.'
+                f'File:{abs_path} not read. Check the name and path.')
+
+        mask = (self.ls_df['layer'] == layer_number) & (self.ls_df['datatype']
+                                                        == datatype)
+
+        search_result_df = self.ls_df[mask]
+
+        if len(search_result_df) > 0:
+            try:
+                thickness = self.multi_planar_design.parse_value(
+                    search_result_df.thickness.iloc[0].strip('\''))
+                z_coord = self.multi_planar_design.parse_value(
+                    search_result_df.z_coord.iloc[0].strip('\''))
+                material = search_result_df.material.iloc[0].strip('\'')
+                chip_name = search_result_df.chip_name.iloc[0].strip('\'')
+                value = search_result_df.fill.iloc[0].strip('\'')
+                if value in TRUE_STR:
+                    fill_value = True
+                elif value in FALSE_STR:
+                    fill_value = False
+                else:
+                    self.logger.warning(
+                        f'The \"fill\" value is neither True nor False.'
+                        f'You have:{value}.  '
+                        f'Will return NULL for fill value.')
+                props['thickness'] = thickness
+                props['z_coord'] = z_coord
+                props['material'] = material
+                props['fill'] = fill_value
+                props['chip_name'] = chip_name
+            except Exception as ex:
+                self._warning_search_minus_chip(layer_number, datatype, ex)
+        result = list()
+        for item in properties:
+            result.append(props[item])
+        return tuple(result)
+
+    def is_layer_unique(self) -> bool:
+        """Check to sort on layer number make sure they are unique.
+        This method is not so relevant, since layers can have datatype entries.
+        Thus there can be rows with same layer number.
+        """
+        return self.ls_df['layer'].is_unique
+
+    def _read_csv_df(self, abs_path: str) -> None:
+        #ASSUME that self.filename_csv_df is valid file path and name.
+        #https://best-excel-tutorial.com/59-tips-and-tricks/485-how-to-display-a-single-quote-in-a-cell
+        try:
+            self.ls_df = pd.read_csv(abs_path)
+
+        except BaseException as error:
+            self.logger.warning(
+                f'Not able to create pandas dataframe.'
+                f'File:{abs_path} not imported. Expected a CSV formatted file.')
+            self.logger.error(f'The exception: {error}')
+
+    def get_fill_for_chip_layer_datatype(self,
```

Comment:
Is there a need of a separate function for getting fill? Looks like this is already covered in `get_properties_for_layer_datatype` as layer is always unique...

Reply:
I was attached to this method, but yes, we can remove this and will update my code which uses it.